static/js/app.js

```diff
@@ -0,0 +1,26 @@
+document.getElementById('last-generated').textContent =
+'Last generated ' + timeDifference(new Date().getTime(), {{timestamp}})
+
+function timeDifference(current, previous) {
+  const msPerMinute = 60 * 1000
+  const msPerHour = msPerMinute * 60
+  const msPerDay = msPerHour * 24
+  const msPerMonth = msPerDay * 30
+  const msPerYear = msPerDay * 365
+
+  const elapsed = current - previous
+
+  if (elapsed < msPerMinute) {
+    return Math.round(elapsed / 1000) + ' seconds ago'
+  } else if (elapsed < msPerHour) {
+    return Math.round(elapsed / msPerMinute) + ' minutes ago'
+  } else if (elapsed < msPerDay) {
+    return Math.round(elapsed / msPerHour) + ' hours ago'
+  } else if (elapsed < msPerMonth) {
+    return 'approximately ' + Math.round(elapsed / msPerDay) + ' days ago'
+  } else if (elapsed < msPerYear) {
+    return 'approximately ' + Math.round(elapsed / msPerMonth) + ' months ago'
+  } else {
+    return 'approximately ' + Math.round(elapsed / msPerYear) + ' years ago'
+  }
+}
```

Comment:
We should probably be running tests on this in the `.coafile`. However they might differ from the other JavaScript tests because it's browser and not Node.js.

Reply:
Now that you talk about it, I forgot to told you to put real linting rules :P e.g. `eslint: recommended`